apps/desktop/src/app/main-layout/sidebar.component.html

```diff
@@ -1,59 +1,69 @@
-          {{ text | translate }}
-        </span>
-        <div
-          *ngIf="step === 0 && (currentStep | async) === 0"
-          class="quick-access-list">
-          <div
-            *ngFor="let item of cache | async"
-            class="quick-access">
+            class="step"
+            [class.pending]="(currentStep | async) < step"
+            [class.active]="(currentStep | async) === step"
+            [class.clickable]="step === 0"
+            (click)="goToStep(step)">
             <span
-              (click)="goToSource(item.connectorId, item.name, false)"
-              paEllipsisTooltip>
-              {{ sources[item.connectorId].definition.title + ' - ' + item.name }}
+              *ngIf="(currentStep | async) <= step"
+              class="step-number">
+              <pa-icon name="circle-dash"></pa-icon>
+              <span>{{ step + 1 }}</span>
             </span>
-            <pa-button
-              (click)="goToSource(item.connectorId, item.name, true)"
-              size="small"
-              iconSize="small"
-              icon="edit"
-              aspect="basic"></pa-button>
+            <pa-icon
+              *ngIf="(currentStep | async) > step"
+              name="circle-check"
+              color="#f06"></pa-icon>
+            {{ text | translate }}
+          </span>
+          <div
+            *ngIf="step === 0 && (currentStep | async) === 0"
+            class="quick-access-list">
+            <div
+              *ngFor="let item of sources | async"
+              class="quick-access">
+              <span
+                (click)="goToSource(item.connectorId, item.name, false)"
+                paEllipsisTooltip>
+                {{ item.name }}
+              </span>
+              <span>
+                <pa-button
+                  (click)="goToSource(item.connectorId, item.name, true)"
+                  size="small"
+                  iconSize="small"
+                  icon="edit"
+                  aspect="basic"></pa-button>
+                <pa-button
+                  (click)="deleteSource(item.name)"
+                  size="small"
+                  iconSize="small"
+                  icon="trash"
+                  aspect="basic"></pa-button>
+              </span>
+            </div>
           </div>
-        </div>
-      </li>
-    </ul>
-  </div>
+        </li>
+      </ul>
+    </div>
 
-  <div class="history-menu">
-    <ul>
-      <li>
-        <span (click)="goToHistory()">{{ 'history.title' | translate }}</span>
-      </li>
+    <div class="history-menu">
       <ul>
-        <li *ngIf="hasActiveUploads | async">
-          <span (click)="goToHistory(true)">{{ 'history.active' | translate }}</span>
+        <li>
+          <span (click)="goToHistory()">{{ 'history.title' | translate }}</span>
         </li>
+        <ul>
+          <li>
+            <span (click)="goToHistory(true)">{{ 'history.active' | translate }}</span>
+          </li>
+        </ul>
       </ul>
```

Comment:
this list within a list with only one `li` each seems super weird to me. What do you try to achieve here?

Reply:
That was the existing structure, I have only changed the `(click)` here, but my understanding is it is a trick to implement the indendation.

Reply:
this structure is a non sense. This is semantically wrong and can be easily achieved using pure css.